src/supported-operations/principles.md

```diff
@@ -0,0 +1,57 @@
+## Proposal 
+
+* An operation is implemented by an executable that is responsible for:
+  * the interactions with the cloud (using the locally bridged MQTT topics),
+  * requesting any required parameters,
+  * reporting the operation progress,
+  * returning any expected results.
+* On installation, an operation is declared to thin-edge using a configuration file put in an operation directory.
+  * This directory is organized in sub-directories per cloud and per operation.
+  * Each operation is represented by a configuration file (using the TOML file format).
+  * ```shell
+    $ ls -l /etc/tegde/operations/c8y
+    -rw-rw-r-- 1 user user    688 Jan 1 00:01 c8y_LogfileRequest
+    -rw-rw-r-- 1 user user    331 Jan 1 00:01 c8y_SoftwareUpdate
+    -rw-rw-r-- 1 user user     40 Jan 1 00:01 c8y_Restart
+    ```
+* An operation might run independently of thin-edge.
+  * In that case, the operation plugin has to run as a daemon (listening for requests)
+    and ensure that this daemon is enabled of device re-start.
+  * The operation daemon is responsible for triggering the operations.
+  * For thin-edge, the operation just needs to be declared using an empty TOML file. 
+  * Thin-edge is only responsible for notifying that the operation is available.
+* An operation might be executed on request.
+  * Thin-edge is then responsible for listening for requests and spawning processes to handle these.
+  * The operation configuration provide the topic and the pattern matching the awaking events.
+  * Which command to run is specified by the plugin configuration file.
+  * Similarly, the configuration file specifies on behalf of which user the plugin command has to be run.
+  * ```toml
+    [exec]
+    topic = "c8y/s/ds"
+    on_message = "522,*"
+    command = "/etc/tedge/plugins/c8y_LogfileRequest"
+    user = "root"
```

Comment:
Operations like `c8y_LogfileRequest` and `c8y_Configuration` requires additional MQTT messages to be sent from the device to the cloud with the supported log types(SR 118) and configuration types(SR 119). Would those messages be added as configuration elements here or we expect the script to send that directly via MQTT?

Reply:
I think it's simpler to let the operation plugin sends this messages. The main reason is that the mapper cannot model all the current and future interactions between the device and the cloud operation handlers.

This raised an important point though. *When will the operation plugin be given the opportunity to send these messages?*. One needs a command to run on mapper restart, an `init` command for the plugin.